File: problems/boj/2476.js
let sum = 0

require('fs')
  .readFileSync('/dev/stdin')
  .toString()
  .split('\n')
  .slice(1)
  .forEach((input) => {
    const parse = input.split(' ').map((s) => parseInt(s))
    const r = [...new Set(parse)]

    let ans = 0

    if (r.length === 3) {
      ans = Math.max(r) * 100
    } else if (r.length === 2) {
      const max = Math.max(r)
      const s = parse.filter((a) => a === max)
      const m = s.length > 1 ? max : Math.min(r)
      ans = 1000 + m * 100
    } else {
      ans = 10000 + Math.max(r) * 1000
    }

    sum = ans >= sum ? ans : sum
  })

console.log(sum)

File: problems/boj/2476.py
T = int(input())
ans = []


def remove_duplicate(l):
  return [*set(l)]


for i in range(T):
  a, b, c = map(int, input().split(' '))
  no_dup = remove_duplicate([a, b, c])

  if len(no_dup) == 2:
    who_dup = 0
    for j in no_dup:
      if [a, b, c].count(j) > 1:
        who_dup = j
    ans.append(1000 + who_dup * 100)
  elif len(no_dup) == 3:
    top = sorted([a, b, c])[2]
    ans.append(top * 100)
  else:
    ans.append(10000 + a * 1000)

print(sorted(ans)[T - 1])